dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionService.java

```diff
@@ -99,19 +102,19 @@ private boolean updateContributionsAndCreateFile(Map<String, CONTRIBUTIONS> succ
                 log.info("Contributions failed to send: {}", failedContributions.size());
             }
             try {
-                fileSentSuccess = contributionUpdateRequest(xmlFile, successfulIdList, successfulIdList.size(),fileName,ackXml);
+                contributionFileId = contributionUpdateRequest(xmlFile, successfulIdList, successfulIdList.size(),fileName,ackXml);
             }
-            catch (HttpServerErrorException e){
+            catch (MaatApiClientException | WebClientException| HttpServerErrorException e){
                 // If failed, we want to handle this. As it will mean the whole process failed for current day.
                 log.error("Contributions file failed to send! Investigation needed. State of files will be out of sync!");
                 // TODO: Need to figure how we're going to log a failed call to the ATOMIC UPDATE.
                 throw e;
             }
         }
-        return fileSentSuccess;
+        return contributionFileId != null;
```

Comment:
Can use Objects.nonNull(contributionFileId) if you want.

Reply:
@LAA-Dave-Edwards In the long-term I was thinking it should return the Integer contributionId or throw (again I originally did this, but undid it to keep the patch focussed ....). I'll leave it like this to remind me of that unless you strongly prefer `nonNull`